.github/workflows/reusable-pre_commit.yml

```diff
@@ -41,9 +41,24 @@ jobs:
       SKIP: ${{ inputs.skip }}
     timeout-minutes: ${{ inputs.timeout-minutes }}
     steps:
+      - name: Get base depth
+        id: base-depth
+        if: ${{ (github.event_name == 'pull_request') }}
+        run: |
+          depth=$(expr ${{ github.event.pull_request.commits }} + 1)
+          echo "base-depth=$depth"
+          echo "base-depth=$depth" >> $GITHUB_OUTPUT
       - uses: actions/checkout@v4
-      - run: git fetch --deepen=500 origin ${{github.base_ref}}
         if: ${{ (github.event_name == 'pull_request') }}
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
is it required to force ref? is not default supposed to pick the current one that trigger the build?